lib/QoreClass.cpp

```diff
@@ -561,12 +568,19 @@ qore_class_private::qore_class_private(const qore_class_private& old, QoreProgra
     }
 
     // copy member list
-    for (QoreMemberMap::DeclOrderIterator i = old.members.beginDeclOrder(), e = old.members.endDeclOrder(); i != e; ++i)
-        members.addNoCheck(strdup(i->first), i->second ? i->second->copy(i->first, this) : 0);
+    for (QoreMemberMap::DeclOrderIterator i = old.members.beginDeclOrder(), e = old.members.endDeclOrder(); i != e; ++i) {
+        members.addNoCheck(strdup(i->first), i->second ? new QoreMemberInfo(*i->second, this) : nullptr);
+    }
 
     // copy static var list
-    for (QoreVarMap::DeclOrderIterator i = old.vars.beginDeclOrder(), e = old.vars.endDeclOrder(); i != e; ++i)
-        vars.addNoCheck(strdup(i->first), i->second ? i->second->copy(i->first) : 0);
+    for (QoreVarMap::DeclOrderIterator i = old.vars.beginDeclOrder(), e = old.vars.endDeclOrder(); i != e; ++i) {
+        vars.addNoCheck(strdup(i->first), i->second ? i->second->copy(i->first) : nullptr);
```

Comment:
Could it also be `new QoreMemberInfo(*i->second)` instead of `i->second->copy(i->first)` or am I missing something?

Reply:
also done